sources/rest_api/__init__.py

```diff
@@ -0,0 +1,768 @@
+
+        if len(resolved_params) > 1:
+            raise ValueError(
+                f"Multiple resolved params for resource {resource_name}: {resolved_params}"
+            )
+
+        predecessors = set(x.resolve_config.resource_name for x in resolved_params)
+
+        dependency_graph.add(resource_name, *predecessors)
+        endpoint_resource["_resolved_param"] = (
+            resolved_params[0] if resolved_params else None
+        )
+
+        if resource_name in endpoint_resource_map:
+            raise ValueError(f"Resource {resource_name} has already been defined")
+
+        endpoint_resource_map[resource_name] = endpoint_resource
+
+    # Create the resources
+    for resource_name in dependency_graph.static_order():
+        endpoint_resource = endpoint_resource_map[resource_name]
+        endpoint_config = endpoint_resource["endpoint"]
+        request_params = endpoint_config.get("params", {})
+        paginator = create_paginator(endpoint_config.get("paginator"))
+
+        # TODO: Remove _resolved_param from endpoint_resource
+        resolved_param: ResolvedParam = endpoint_resource.pop("_resolved_param", None)
+        include_from_parent: list[str] = endpoint_resource.pop(
+            "include_from_parent", []
+        )
+        if not resolved_param and include_from_parent:
+            raise ValueError(
+                f"Resource {resource_name} has include_from_parent but is not "
+                "dependent on another resource"
+            )
+
+        resource_kwargs = remove_key(endpoint_resource, "endpoint")
+
+        incremental_object, incremental_param = setup_incremental_object(
+            request_params, endpoint_config.get("incremental")
+        )
+
+        if resolved_param is None:
+
+            def paginate_resource(
+                method,
+                path,
+                params,
+                paginator,
+                incremental_object=incremental_object,
+                incremental_param=incremental_param,
+            ):
+                if incremental_object:
+                    params[incremental_param] = incremental_object.last_value
+
+                yield from client.paginate(
+                    method=method,
+                    path=path,
+                    params=params,
+                    paginator=paginator,
+                )
+
+            resources[resource_name] = dlt.resource(
+                paginate_resource, **resource_kwargs
+            )(
+                method=endpoint_config.get("method", "get"),
+                path=endpoint_config.get("path"),
+                params=request_params,
+                paginator=paginator,
+            )
+
+        else:
+            predecessor = resources[resolved_param.resolve_config.resource_name]
+
+            param_name = resolved_param.param_name
+            request_params.pop(param_name, None)
+
+            def paginate_dependent_resource(
```

Comment:
I wonder if this function could be top level and not inner? so maybe people can mock their own in worst case?